backend/dal/Services/Concrete/ParcelService.cs

```diff
@@ -354,42 +359,58 @@ public Parcel Update(Parcel parcel)
                 }
 
                 // Delete any missing records in child collections.
-                foreach (var building in originalParcel.Buildings)
+                foreach (var parcelBuilding in originalParcel.Buildings)
                 {
-                    var matchingBuilding = parcel.Buildings.FirstOrDefault(b => b.Id == building.Id);
-                    if (matchingBuilding == null)
+                    var matchingParcelBuilding = parcel.Buildings.FirstOrDefault(pb => pb.BuildingId == parcelBuilding.BuildingId);
+                    if (matchingParcelBuilding == null)
                     {
-                        this.ThrowIfNotAllowedToUpdate(building, _options.Project);
-                        this.Context.Buildings.Remove(building);
+                        // TODO: This will delete the building from the datasource.  This isn't entirely ideal as it may be related to multiple parcels.
+                        this.ThrowIfNotAllowedToUpdate(matchingParcelBuilding.Building, _options.Project);
+
+                        var parcelBuildings = this.Context.ParcelBuildings.Where(pb => pb.BuildingId == parcelBuilding.BuildingId).ToArray();
+                        parcel.Buildings.Remove(matchingParcelBuilding);
```

Comment:
Not crazy about the naming of the join table here, but there are bigger fish.

Reply:
This is just a standard many-to-many naming conventions, Parcel -> ParcelBuilding -> Building

Reply:
I meant the Parcel.Buildings.Buildings